enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,279 @@
+  - "@dhellmann"
+creation-date: 2022-03-03
+last-updated: 2022-03-03
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/AGENT-9
+---
+
+# Agent-based Installer
+
+## Summary
+
+The agent-based installer is an installation method that will use a bootable,
+ephemeral installer image running on the hosts that are to become part of the
+cluster. The user will generate the image using a command-line tool. The image
+itself will contain components (such as
+[assisted-service](https://github.com/openshift/assisted-service/#readme) and
+[assisted-agent](https://github.com/openshift/assisted-installer-agent#readme))
+of the Assisted and Multi-Cluster Engine (MCE) installation methods. The
+installer will be usable in both a fully-automated workflow, where
+configuration is provided upfront, and an interactive workflow where the user
+can interact with a GUI similar to the hosted Assisted installation service.
+
+## Motivation
+
+Every on-premises user has a first cluster ("Cluster 0"), and many of them will
+be installed in a disconnected environment. While powerful tools exist for
+creating multiple on-premises clusters (Hive, HyperShift, MCE), they all run in
+a cluster themselves, which must somehow be provisioned without these tools.
+
+Each of the current installation options has limitations that make it
+unpalatable to a subset of users. IPI is highly opinionated, and suffers from
+providing limited visibility of configuration problems in on-premises
+environments. UPI exposes users to the full complexity of OpenShift to the user
+with little guidance. The (hosted) Assisted service is not available for
+IPv6-only or disconnected environments. For some users, none of these options
+is optimal.
+
+On-premises users need a simple way to deploy their first cluster, even in a
+disconnected environment, using their own automation to boot hosts.
+
+### Goals
+
+- Install clusters in fully-disconnected environments.
+- Make use of assisted installation components to pre-validate the installation
+  environment.
+- Require no machines in the cluster environment other than those that are to
+  become part of the cluster.
+- Install clusters with single-node, compact, and high-availability topologies.
+- Be agnostic to the tools used to provision (virtual) machines.
+- Support install configurations with platform `none`, `baremetal`, or
+  `vsphere`.
+
+### Non-Goals
+
+- Replace the IPI installer in any capacity.
+- Automate booting of the image on (virtual) machines.
+- Support install configurations for cloud-based platforms.
+- Provide an interactive API to drive automated deployments.
+- Install minor versions of OpenShift that are older than the agent-based
+  installer version.
+- Operate in disconnected environments without the need to separately mirror
+  the release payload to a registry that is usable from the installed cluster.
```

Comment:
We have some future goals that could be called out as non-goals for this enhancement that would be covered by other enhancements later

- Support multiple image formats
- Create minimal bootable images and separate root filesystems

Reply:
Why not? Isn't it built in already? Ipxe and minimal iso is there. I would say that releasing without minmal iso is sbuoptimal

Reply:
It's a good point. Is it acceptable releasing without minimal iso support? 
@julienlim

Reply:
AIUI users can build a minimal ISO from the ISO using coreos-installer, and they can certainly build PXE artifacts. We might want to document this but I don't think we have to support it directly in the tool to have a viable installer.

Reply:
Right, remember this doc isn't the only place we'll talk about work on this project. Right now we're not focusing on multiple image formats. We know we'll need to, and we'll want a separate document when we do.